src/packaging/_types.py

```diff
@@ -0,0 +1,46 @@
+"""Types used internally.
+
+This module defines types separately so that there is no runtime cost.
+"""
+from __future__ import annotations
+
+from typing import Any, Callable, List, NewType, Sequence, Tuple, TypeVar, Union
+
+from ._parser import Op, Value, Variable
+from ._structures import InfinityType, NegativeInfinityType
+from .version import Version
+
+MarkerVar = Union[Variable, Value]
+MarkerItem = Tuple[MarkerVar, Op, MarkerVar]
+# MarkerAtom = Union[MarkerItem, List["MarkerAtom"]]
+# MarkerList = List[Union["MarkerList", MarkerAtom, str]]
+# mypy does not support recursive type definition
+# https://github.com/python/mypy/issues/731
```

Comment:
I don't think this is valid anymore. The issue is closed in the mypy repo, maybe you could check? WOuld be nice to avoid hauling around the tech debt.

Reply:
I thought the same thing and tried but encountered errors.